airbyte-integrations/connectors/source-twilio-taskrouter/source_twilio_taskrouter/twilio_taskrouter.yaml

```diff
@@ -0,0 +1,70 @@
+version: "0.1.0"
+
+
+definitions:
+  selector:
+    extractor:
+      field_pointer: []
+  requester:
+    url_base: "https://taskrouter.twilio.com"
+    http_method: "GET"
+    authenticator:
+      type: BasicHttpAuthenticator
+      username: "{{ config['account_sid'] }}"
+      password: "{{ config['auth_token'] }}"
+  retriever:
+    record_selector:
+      $ref: "*ref(definitions.selector)"
+    requester:
+      $ref: "*ref(definitions.requester)"
+  base_stream:
+    retriever:
+      $ref: "*ref(definitions.retriever)"
+  allworkspaces_stream:
+    $ref: "*ref(definitions.base_stream)"
+    $options:
+      name: "allworkspaces"
+      path: "/v1/Workspaces"
+      primary_key: "sid"
+    retriever:
+      $ref: "*ref(definitions.base_stream.retriever)"
+      record_selector:
+        extractor:
+          field_pointer: ["workspaces"]
+      paginator:
+        type: "DefaultPaginator"
+        page_size_option:
+          inject_into: "request_parameter"
+          field_name: "page_size"
+        pagination_strategy:
+          type: "CursorPagination"
```

Comment:
the paginator doesn't turn to the next page. In my test account, I have 57 workers, I would expect the connector firstly fetches the first 50 workers as 50 is the default page size, and it increases the page to 1 and use `page=1` to get the remaining 7.  But only the first 50 are read. 

```
{"type": "LOG", "log": {"level": "INFO", "message": "Read 50 records from workers stream"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Finished syncing workers"}}
{"type": "LOG", "log": {"level": "INFO", "message": "SourceTwilioTaskrouter runtimes:\nSyncing stream allworkspaces 0:00:00.364470\nSyncing stream workers 0:00:00.780400\nSyncing stream workspaces 0:00:00.384402"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Finished syncing SourceTwilioTaskrouter"}}
```

Reply:
I've made sure the edit access for maintainers is checked, just need to fix the paginator. What could be the reason for it to be unable to read the remaining records. I'll try testing it out by creating more on my test acc

Reply:
can you run the test again and see if it works?

Reply:
> What could be the reason for it to be unable to read the remaining records

The reason is that you need to parse response.body.meta, get the page number in the `next_page_url`, and use it to send the request again, until next_age_url is null. 

```
request_url = "base_url/v1/workers";
do {
  response = request.get(request_url);
  request_url = response.body.meta.next_page_url;
} while (request_url == null)
```

The pseudo-code looks like above.

Reply:
where would i need to enter the code? in which file specifically?

Reply:
You can run the same read in your local machine. I listed the command above. You don't have to create so many workers, you can test it with a a smaller page_size. For example, if you have 5 workers, and the page_size is 2, the last one worker appears in page 2. If you see the log of `Read 5 records from workers stream`, you are good. 

I can't always stay in your topic branch, I am working on another connector. Please verify it in your test account first. Thanks.

Reply:
Since a new paginator is introduced, please run unit tests, integration tests, and acceptance test in your local environment and provide the logs again. The one in the PR description is out-dated. Thanks. 

https://docs.airbyte.com/connector-development/testing-connectors/source-acceptance-tests-reference

Reply:
I tried with 5 workers and a page size of 2 and obtain the desired output 
![image](https://user-images.githubusercontent.com/59562284/200192386-3b7a3457-9cf2-4a96-8a82-af49c7dbbd92.png)

Reply:
Here is the log of the new sets of tests
![image](https://user-images.githubusercontent.com/59562284/200192527-7296fb61-06b9-4db2-80fc-6c1ae13476d2.png)

Thanks for all your help, really appreciate it

Reply:
You are able to fetch 5 workers because the page_size of 2 is not passed to the request parameters. The property in the config.json plays no effects. 

I still don't get 57 workers. When I use a different page_size, the result is the same, meaning only the first page is fetched.

Reply:
> where would i need to enter the code? in which file specifically? Is it by done by defining custom components?

I just offer you a pseudo-code to explain the algorithm behind, you need to set up the paginator correctly so that it works like what the algorithm describes. According to your yaml file, your paginator doesn't parse the response at all. Check greenhouse.yaml for an example.